waku/v2/node/config.nim

```diff
@@ -46,6 +46,11 @@ type
       desc: "Specify method to use for determining public address. " &
             "Must be one of: any, none, upnp, pmp, extip:<IP>."
       defaultValue: "any" }: string
+
+    maxConnections* {.
+      desc: "Maximum allowed number of libp2p connections."
+      defaultValue: 50
+      name: "max-connections" }: uint16
```

Comment:
Is there any specific reason for `uint16` type? I see a conversion to `int` in the `WakuNode.new(...` (`conf.maxConnections.int`) for the max connection, what do you think about defaulting it to `int`? unsigned values are prone to computational errors

Reply:
The underlying `switch` builder requires an `int`, hence the conversion. In general, though, we want to avoid using `int`, as its size [varies according to platform](https://status-im.github.io/nim-style-guide/language.integers.html).
For now, `uint16` is what's used for some other config options (so I tried to stay consistent) and will ensure that people don't accidentally configure a silly max connection value, like a negative value or one that's in the millions (it ensures 0 < `maxConnections` < 65535). LMK if this reasoning makes sense!

Reply:
Ah, right! not `int` but some int with a specific size 👍 

Another possible problem with unsigned int is in subtraction operations where you never get a negative value :P I imagine since the max connection is checked at the p2p level (in which the maxConnection is an `int` type), there should be no issue. Thanks for the explanation! :)